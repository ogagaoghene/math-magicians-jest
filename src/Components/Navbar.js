import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import '../css/navbar.css';

const Navbar = () => {
  return (
    <nav className="navigation">
      <div className="brand-name">
        Math Magicians
      </div>
     
      <div>
        <ul>
          <li><NavLink to="/">Home</NavLink></li>
          <li><NavLink to="/Calculator">Calculator</NavLink></li>
          <li><NavLink to="/Quotes">Quotes</NavLink></li>
        </ul>
      </div>
    </nav>
  );
};

export default Navbar;
