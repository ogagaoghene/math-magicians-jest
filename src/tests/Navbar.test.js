import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../Components/Navbar';

describe('Snapshot testing', () => {
  it('Check if Navbar component is rendered', () => {
    const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
