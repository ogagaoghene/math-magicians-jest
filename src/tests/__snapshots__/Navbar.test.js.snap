// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Check if Navbar component is rendered 1`] = `
<nav
  className="navigation"
>
  <div
    className="brand-name"
  >
    Math Magicians
  </div>
  <div>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/Calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/Quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
